scala/data_structures/trie.sc

```diff
@@ -17,7 +18,7 @@ final case class Trie(
       .map(_.toChar)
       .mkString("")
 
-  def +=(word: String): Trie = {
+  def :+(word: String): Trie = {
```

Comment:
`+=` made it look like we were mutating the structure. `:+` follows Scala's standards.